src/category_theory/limits/shapes/binary_products.lean

```diff
@@ -57,6 +57,11 @@ def map_pair {W X Y Z : C} (f : W ⟶ Y) (g : X ⟶ Z) : pair W X ⟶ pair Y Z :
 @[simp] lemma map_pair_left {W X Y Z : C} (f : W ⟶ Y) (g : X ⟶ Z) : (map_pair f g).app walking_pair.left = f := rfl
 @[simp] lemma map_pair_right {W X Y Z : C} (f : W ⟶ Y) (g : X ⟶ Z) : (map_pair f g).app walking_pair.right = g := rfl
 
+/-- Every functor indexing a (co)product is naturally isomorphic (actually, equal) to a `pair` -/
+def pair_of_diagram {F : discrete walking_pair ⥤ C} :
```

Comment:
I'm slightly squeamish about having this argument implicit. I see you can get away with it so far, but I think it's sufficiently inexpensive that you ought to make this an explicit argument.

Reply:
Agreed.